### ISSUE TITLE
Chat middleware can choose to "squelch" a message

### ISSUE THREAD

**Issue by mfvargo** [actionhero/actionhero#1973]
**Description of the feature**
If a chat room "say" or "onSayReceive" middleware function returns null, action hero should "squelch" the message.  This means the message should no longer be processed.  In the case of the onSayReceive the message should not be published to redis.  In the "say" case, the particular connection on which the middleware is being called should not get the message sent to it.

Some applications would benefit by allowing the chat middleware to "squelch" a message (either entirely, or selectively by room member).  My specific example is a IOT "Meet me in the middle" bridge to connect IOT devices to web apps.  The devices do not need to receive back an echo of the messages they send (save bandwidth and processing).  Also command messages from the web app to the IOT device do not need to be echoed back to the web app sending the message.

The squelch on onSayReceive might be used to prevent malformed messages from being distributed to the room.  Maybe some kind of first line defense against a DOS attack.

**Describe the solution you'd like**
If the middle ware callback returns null.  discontinue processing of that message.

**Describe alternatives you've considered**
One possibility is to have the middleware just return {} which would still send a websocket message but at least would reduce the overhead of that mesasge.  room clients would need to make sure to ignore an "empty" message

**Comment by mfvargo** [comment]
working on this now.  Code is pretty straightforward.  Trying to figure out where to put tests.  I'm guessing __tests__/core/middleware.ts

**Comment by mfvargo** [comment]
You can kind of achieve this today by having the middleware throw an exception.  Works as you would want but it does cause an log message to get spit out which might not be the best.
```
2021-09-30T23:17:54.602Z - warning: Error: Echo Squelched messagePayload=[object Object] connection=[object Object] 
```

**Comment by evantahler** [comment]
Yeah, you probably don't want to throw :D 

You can test this along with the middleware, and in the websocket server tests https://github.com/actionhero/actionhero/blob/main/__tests__/servers/websocket.ts#L262

**Comment by mfvargo** [comment]
This was merged into 27.1 so I think this can be closed.
